app/screens/code/index.tsx

```diff
@@ -19,16 +25,41 @@ type Props = {
     textStyle: TextStyle;
 }
 
+const COPY_CODE_BUTTON = 'copy-code';
+
 const edges: Edge[] = ['left', 'right'];
 
 const styles = StyleSheet.create({
     flex: {flex: 1},
 });
 
 const Code = ({code, componentId, language, textStyle}: Props) => {
+    const theme = useTheme();
     const managedConfig = useManagedConfig<ManagedConfig>();
     useAndroidHardwareBackHandler(componentId, popTopScreen);
 
+    const copyToClipboard = useCallback(() => {
+        if (!code) {
+            return;
+        }
+
+        Clipboard.setString(code);
+        showSnackBar({barType: SNACK_BAR_TYPE.CODE_COPIED, sourceScreen: componentId});
+    }, [code, componentId]);
+
+    useNavButtonPressed(COPY_CODE_BUTTON, componentId, copyToClipboard, []);
```

Comment:
This may be a oversight on the hook definition, or some optimization we did, but you need to pass everything needed to the dependency array. Since you are passing nothing, any change in the `copyToClipboard` function would not be reflected in the button. Probably this is "fine" right now, because the code is not supposed to change while this screen is open, but still, let's add it for future-proofing.

And just for good measure (even though it should be constant through the lifetime of the component) add also the componentId.